File: backend/src/main/java/bsuir/service/StationDetails/TrainService.java
package bsuir.service.StationDetails;

import bsuir.model.stationDetails.Train;

import java.util.List;


public interface TrainService {
    Train save(Train train);

    void delete(long id);

    Train getById(long id);

    List<Train> getAll();

    List<Train> getAllByModel(long model);
}

File: frontend/src/app/services/station-details/train.service.ts
import {Injectable} from '@angular/core';
import {HttpClient, HttpParams} from "@angular/common/http";
import {City} from "../../models/station-details/city";
import {Observable} from "rxjs";
import {Train} from "../../models/station-details/train";

@Injectable({
  providedIn: 'root'
})
export class TrainService {

  private url: string = "/api/trains";

  constructor(private http: HttpClient) {
  }

  public save(train: Train): Observable<Train> {
    return this.http.post<Train>(this.url + '/save', train);
  }

  public getAll(): Observable<Train[]> {
    return this.http.get<Train[]>(this.url + '/get-all');
  }

  public getAllByModel(model: number): Observable<Train[]> {
    return this.http.get<Train[]>(
      this.url + '/by-model',
      {
        params: new HttpParams()
          .set('model', model.toString())
      })
  }
}
